### PR TITLE
Koala — Round 2

### PR DESCRIPTION
**Pull request by GawainLynch**
@SahAssar you are far more familiar with the functionality, so please let me know if something isn't working as expected (it will also probably then hit us internally :wink:), and I will get onto it.

I killed off a few things in the request handling by pushing the local selection to a service, but I think the original assumptions were made on things "Not Quite Right" in Bolt 2.

### CODE REVIEW

File: src/Config/Config.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return Locale
+     */
+    public function getLocale($name)
+    {
+        return $this->get('locales')->get($name);
```
Comment (SahAssar) [inline src/Config/Config.php]: `$this->get('locales')` returns an array, right?

So the second `get()` will not work, right?

Comment (GawainLynch) [inline src/Config/Config.php]: Whoops… thought I caught that in a rebase… my bad

**Comment by GawainLynch** [comment]
…and with the magic of `git --rebase` that never happened :wink:

File: templates/fields/_locale.twig [hunk 2]
```diff
@@ -10,8 +10,8 @@
 {% endfor %}
 
 {% for key, option in locales %}
-    {% set current = option.slug == app.request.get('_locale') ? option : current %}
-    {% set currentkey = option.slug == app.request.get('_locale') ? key : currentkey %}
+    {% set current = option.get.slug == app.request.get('_locale') ? option.get.slug : current %}
```
Comment (SahAssar) [inline templates/fields/_locale.twig]: Shouldn't these be `option.get('slug')` (or `option.slug` works too) rather than option.get.slug?

Comment (GawainLynch) [inline templates/fields/_locale.twig]: White :koala:s can't jump

**Comment by GawainLynch** [comment]
…and with the magic of `git --rebase` that **also** never happened :wink:

File: templates/fields/_locale.twig [hunk 3]
```diff
@@ -30,8 +30,8 @@
                 <li>
                     <a href="{{ url(
                                 app.request.get('_route'),
-                                app.request.get('_route_params')|merge({_locale: option.slug })
-                            )}}">{{ option.label }}</a>
+                                app.request.get('_route_params')|merge({_locale: option.get('slug') })
+                            )}}">{{ option.get.label }}</a>
```
Comment (SahAssar) [inline templates/fields/_locale.twig]: ;)

**Comment by GawainLynch** [comment]
…and with the magic of `git --rebase` that **also** **also** never happened :wink:

**Comment by SahAssar** [comment]
Merging in and I'll keep working on some parts of it :)

**Pull request merged**